### PR TITLE
Fix for #957 Requesting permissions is not done everywhere 

### LINKED ISSUE #957: Requesting permissions is not done everywhere
While making PR #956 i noticed not all fragments requested permission when the permission was not already given. After looking a bit into it, the analysis showed me that the following classes are NOT asking for storage/camera permissions when they need:
- HistoryFragment
- InvertPdfFragment
- MergeFilesFragment
- PdfToImageFragment
- QrBarcodeScanFragment (camera gets asked, but not storage for saving)
- RemoveDuplicatePagesFragment
- RemovePagesFragment
- SplitFilesFragment

### Steps to reproduce

1. Remove storage/camera permissions for the app
  a. hold the icon long
  b. click the small i in the top right corner
  c. click on "Permissions" (or how its named)
  d. turn off storage/camera
2. Open the app
3. click on any of the listed fragments
4. try to do something that requires the permission, for example try to open a PDF in the SplitFilesFragment

### Expected behaviour
A permission request should show up asking for storage permission.

### Actual behaviour
The snackbar shows up with the R.string.encrypted_pdf message.

### Would you like to work on the issue?
I would work on the issue if no one else wants to do it and i improved the code climate of ImagesToPdfFragment to something better than F.

PS: the behaviour of #881 can not only be caused by the Parent folder not existing but also by this

### PR DESCRIPTION
**Pull request by KAbhijeet2105**
# Description

The issue is fixed by adding permissions(Storage and Camera) requests in the following fragments : 

- HistoryFragment

- InvertPdfFragment

- MergeFilesFragment

- PdfToImageFragment

- QrBarcodeScanFragment 

- RemoveDuplicatePagesFragment

- RemovePagesFragment

- SplitFilesFragment

 
Fixes #957 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Remove storage/camera permissions for the app
     1.hold the icon long
     2.click the small I in the top right corner
     3.click on "Permissions" (or how it's named)
     4.turn off storage/camera

- Open the app

Click on any of the listed fragments
try to do something that requires the permission, for example, try to open a PDF in the SplitFilesFragment

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.6.0-rc01'
+        classpath 'com.android.tools.build:gradle:4.0.1'
```
Comment (Swati4star) [inline build.gradle:12]: revert changes in thsi file

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Oct 06 17:49:52 BRT 2020
+#Sat Oct 17 17:40:10 IST 2020
```
Comment (Swati4star) [inline gradle/wrapper/gradle-wrapper.properties:1]: revert changes in this file too

File: app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java [hunk 3]
```diff
@@ -58,6 +60,8 @@
     private BottomSheetUtils mBottomSheetUtils;
     private static final int INTENT_REQUEST_PICKFILE_CODE = 10;
     private BottomSheetBehavior mSheetBehavior;
+    private static final int PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT = 1;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java:63]: instead of defining this constant everywhere, let's have one in the PermissionUtils

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 4]
```diff
@@ -721,6 +721,7 @@ private void getRuntimePermissions() {
                 REQUEST_PERMISSIONS_CODE);
     }
 
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java:724]: revert this change

**Review by Swati4star — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (KAbhijeet2105) [reply build.gradle:12]: After reverting the changes : 
Build failed due to: The project is using an incompatible version (3.6.0-rc01) of the Android Gradle plugin.

File: app/build.gradle [hunk 5]
```diff
@@ -64,6 +64,9 @@ dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
     implementation 'junit:junit:4.12'
     implementation 'com.android.support:appcompat-v7:28.0.0'
+    implementation 'com.android.support:animated-vector-drawable:28.0.0'
+    implementation 'com.android.support:exifinterface:28.0.0'
```
Comment (Swati4star) [inline app/build.gradle:68]: why do we need these?

File: app/src/main/java/swati4star/createpdf/fragment/HistoryFragment.java [hunk 6]
```diff
@@ -207,4 +214,15 @@ protected void onPostExecute(Void aVoid) {
...
+    /**
+     * check runtime permissions for storage and camera
+     * */
+    private void getRuntimePermissions() {
+        PermissionsUtils.getInstance().requestRuntimePermissions(this,
+                READ_WRITE_CAMERA_PERMISSIONS,
+                PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT);
+    }
+
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HistoryFragment.java:227]: remove extra new lines

File: app/src/main/java/swati4star/createpdf/fragment/InvertPdfFragment.java [hunk 7]
```diff
@@ -193,5 +197,15 @@ public void closeBottomSheet() {
...
+
+    /**
+     * check runtime permissions for storage and camera
+     * */
+    private void getRuntimePermissions() {
+        PermissionsUtils.getInstance().requestRuntimePermissions(this,
+                READ_WRITE_CAMERA_PERMISSIONS,
+                PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT);
+    }
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/InvertPdfFragment.java:209]: remove

File: app/src/main/java/swati4star/createpdf/fragment/PdfToImageFragment.java [hunk 8]
```diff
@@ -70,6 +73,7 @@
     private PDFUtils mPDFUtils;
     private String[] mInputPassword;
 
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/PdfToImageFragment.java:76]: ?

File: app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java [hunk 9]
```diff
@@ -20,6 +20,7 @@
  */
 public class PermissionsUtils {
 
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java:23]: ?

File: app/src/main/res/layout/dialog_theme_default.xml [hunk 10]
```diff
@@ -1,13 +1,17 @@
...
 
     <RadioGroup
         android:id="@+id/radio_group_themes"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:orientation="vertical">
+        android:orientation="vertical"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
```
Comment (Swati4star) [inline app/src/main/res/layout/dialog_theme_default.xml:14]: why?

File: app/src/main/res/values-ja/strings.xml [hunk 11]
```diff
@@ -270,7 +270,7 @@
     <string name="share_files">Share Files</string>
     <string name="images_rearranged">Images rearranged</string>
     <string name="compress_pdf">Compress PDF</string>
-    <string name="compress_pdf_prompt">Enter % compression of PDF :</string>
+    <string name="compress_pdf_prompt">Enter %% compression of PDF :</string>
```
Comment (Swati4star) [inline app/src/main/res/values-ja/strings.xml:273]: ??

**Review by Swati4star — CHANGES_REQUESTED** [review]
The travis build is also broken.
Please add only relevant files to this PR. I can see many unrequired changes

**Comment by KAbhijeet2105** [comment]
> The travis build is also broken.
> Please add only relevant files to this PR. I can see many unrequired changes

Extremely sorry. I was not knowing that travis build is broken. I tried to fix the errors I got in ./gradlew assemble check.

**Comment by Swati4star** [comment]
You could check here: https://travis-ci.org/github/Swati4star/Images-to-PDF/builds/736807833

The build is still broken with this PR.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (Swati4star) [reply gradle/wrapper/gradle-wrapper.properties]: still not resolved

**Comment by Swati4star** [comment]
To fix the build, change this to https://github.com/Swati4star/Images-to-PDF/blob/master/.travis.yml#L7 to 29.0.2
That should fix the build issue

**Comment by KAbhijeet2105** [comment]
> To fix the build, change this to https://github.com/Swati4star/Images-to-PDF/blob/master/.travis.yml#L7 to 29.0.2
> That should fix the build issue

@Swati4star  Thanks. This solution fixed the build issue.

File: app/build.gradle [hunk 12]
```diff
@@ -64,6 +64,7 @@ dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
     implementation 'junit:junit:4.12'
     implementation 'com.android.support:appcompat-v7:28.0.0'
+
```
Comment (Swati4star) [inline app/build.gradle:67]: If there are no relevant changes in a file, there is no point adding it to commit.

File: app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java [hunk 13]
```diff
@@ -104,4 +104,4 @@ public void handleRequestPermissionsResult(Activity context, @NonNull int[] gran
         } else
             StringUtils.getInstance().showSnackbar(context, R.string.snackbar_insufficient_permissions);
     }
-}
+}
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java:107]: ??

File: app/src/main/res/layout/dialog_theme_default.xml [hunk 14]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (Swati4star) [inline app/src/main/res/layout/dialog_theme_default.xml:3]: ??

**Comment by KAbhijeet2105** [comment]
> Can you please self review the files once?
> 
> I have already mentioned in my previous multiple comments to not include file which do not have relevant change

Can you please help?  how to remove files from my PR?

**Comment by KAbhijeet2105** [comment]
@Swati4star @McModknower, please guide me here. how can I remove files from my PR? 
Should I close this PR and Open Another one?

**Comment by Swati4star** [comment]
@KAbhijeet2105 You can rectify this PR.
So, there are a few files which do not have a relevant change.
- You can revert the change in the file. 
For eq: In file app/src/main/res/layout/dialog_theme_default.xml, you remove the new line from line num 5, add that new line back. This will ultimately remove the file from commit.
similarly for other file also

**Comment by KAbhijeet2105** [comment]
> @KAbhijeet2105 You can rectify this PR.
> So, there are a few files which do not have a relevant change.
> 
> * You can revert the change in the file.
>   For eq: In file app/src/main/res/layout/dialog_theme_default.xml, you remove the new line from line num 5, add that new line back. This will ultimately remove the file from commit.
>   similarly for other file also

@Swati4star please check my latest commit and let me know if any changes required.

File: app/src/main/res/layout/dialog_theme_default.xml [hunk 15]
```diff
@@ -2,13 +2,11 @@
 <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent">
-
     <RadioGroup
         android:id="@+id/radio_group_themes"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:orientation="vertical">
-
```
Comment (Swati4star) [inline app/src/main/res/layout/dialog_theme_default.xml:11]: Should this change be here?
If you are having issues, please open a clean new PR and have only required files

Comment (KAbhijeet2105) [reply app/src/main/res/layout/dialog_theme_default.xml:11]: Yes. I will open a new PR.
